### DISCUSSION TITLE
"cwd option must be a path to a directory" error from percy snapshot

### DISCUSSION THREAD

**Discussion by ekinoben** [percy/cli#590] [Support] [answered]
I've installed Percy CLI via `npm install --save-dev @percy/cli`, which has added `"@percy/cli": "^1.0.0-beta.69",` to my `package.json`.

I'm trying to run `npx percy snapshot urls.yml --dry-run`.

This firstly prompts me to install the additional `percy` package:

```
Need to install the following packages:
  percy
Ok to proceed? (y) y
```

Any time I subsequently run the command it errors out with:

>  Error: The `cwd` option must be a path to a directory

If I add `--verbose` option, I get:

> Error: Unexpected argument: --verbose

And if I add `--help` option I get:

```
USAGE
  $ percy snapshot [SNAPSHOTDIRECTORY]

ARGUMENTS
  SNAPSHOTDIRECTORY  [default: .] A path to the directory you would like to snapshot
```

This doesn't match the [usage description in the docs](https://github.com/percy/cli/tree/master/packages/cli-snapshot#readme).

Where have I gone wrong here? It's like I've got an... old version, or something?? How do I get a version that matches the docs?

**Comment by Robdel12** [comment, marked as answer]
Hey @ekinoben! When you get this message: 

```
Need to install the following packages:
  percy
Ok to proceed? (y) y
```

`npx` is going out to the registry to install a package named `percy` because it didn't find an executable in the `node_modules` folder with a matching name. The `percy` package is the old `@percy/agent` package that we can't deprecate/change yet since lots of people rely on that still. 

Where are you running `npx percy snapshot urls.yml --dry-run`? Is it the same directory `@percy/cli` was installed in? What happens if you run `./node_modules/.bin/percy snapshot urls.yml --dry-run`?

**Reply by ekinoben** [reply]
Aha. Yes, that fixed it. My `package.json` (and `node_modules`) was in `$HOME` (it's all in docker) but I was running the command from `/tmp`, where I'd put the URLs file. Running `npx percy snapshot /tmp/urls.yml --dry-run` from `$HOME` works. Many thanks, and my apologies for being daft.

**Reply by Robdel12** [reply]
Ah, docker! That make sense. Sorry about that -- this is totally not a great path to fall down. The `percy` package will _eventually_ be `@percy/cli` (once we get most people to upgrade)

**Reply by AlexB-Adv** [reply]
This just solved my issue when attempting to migrate from an old beta to 1.2.1 - thank you!!!
